src/common/commonutils/UrlUtils.c

```diff
@@ -12,7 +12,7 @@ char* UrlEncode(char* target)
 
     int i = 0, j = 0;
     int targetLength = (int)strlen(target);
-    int encodedLength = 3 * targetLength;
+    int encodedLength = 3 * targetLength + 1;

```

Comment:
>3 * targetLength + 1 [](http://example.com/codeflow?start=24&length=20)

Add parenthesis like below (we try to never rely on just operator precedence):

 int encodedLength = (3 * targetLength) + 1;